src/wh_server.c

```diff
@@ -28,8 +30,573 @@
 
 #include "wolfhsm/wh_message.h"
 #include "wolfhsm/wh_message_comm.h"
+#include "wolfhsm/wh_packet.h"
 #include "wolfhsm/wh_server.h"
 
+#ifdef WOLFHSM_SHE_EXTENSION
+const uint8_t WOLFHSM_SHE_KEY_UPDATE_ENC_C[] = {0x01, 0x01, 0x53, 0x48, 0x45,
+    0x00, 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xB0};
+const uint8_t WOLFHSM_SHE_KEY_UPDATE_MAC_C[] = {0x01, 0x02, 0x53, 0x48, 0x45,
+    0x00, 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xB0};
+#define WOLFHSM_SHE_UID_SZ 15
+const uint8_t WOLFHSM_SHE_UID[WOLFHSM_SHE_UID_SZ] = {0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01};
+const uint8_t WOLFHSM_SHE_PRNG_KEY_C[] = {0x01, 0x04, 0x53, 0x48, 0x45, 0x00,
+    0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xB0};
+const uint8_t WOLFHSM_SHE_PRNG_SEED_KEY_C[] = {0x01, 0x05, 0x53, 0x48, 0x45,
+    0x00, 0x80, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0xB0};
+const uint8_t WOLFHSM_SHE_PRNG_EXTENSION_C[] = {0x80, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x01, 0x00};
+uint8_t WOLFHSM_SHE_PRNG_KEY[WOLFHSM_SHE_KEY_SZ];
+enum WOLFHSM_SHE_SB_STATE {
+    WOLFHSM_SHE_SB_INIT,
+    WOLFHSM_SHE_SB_UPDATE,
+    WOLFHSM_SHE_SB_FINISH,
+    WOLFHSM_SHE_SB_SUCCESS,
+    WOLFHSM_SHE_SB_FAILURE,
+};
+uint8_t hsmShePrngState[WOLFHSM_SHE_KEY_SZ];
+uint8_t hsmSheSbState = WOLFHSM_SHE_SB_INIT;
+uint8_t hsmSheCmacKeyFound = 0;
+uint8_t hsmSheRamKeyPlain = 0;
+uint32_t hsmSheBlSize = 0;
+uint32_t hsmSheBlSizeReceived = 0;
+uint32_t hsmSheInitRng = 0;
+/* cmac is global since the bootloader update can be called multiple times */
+Cmac sheCmac[1];
+#endif
+
+#if 0
```

Comment:
Can you can leave out any dormant code for now since we won't be reviewing it.

Reply:
no, want to get this all in the repo ASAP, it will come out as we get it working